engine/classes/Elgg/Notifications/NotificationsService.php

```diff
@@ -172,22 +187,25 @@ public function enqueueEvent($action, $type, $object) {
 			}
 
 			if ($registered) {
-				$this->queue->enqueue(new \Elgg\Notifications\Event($object, $action));
+				$this->queue->enqueue(new SubscriptionNotificationEvent($object, $action));
 			}
 		}
 	}
 
 	/**
 	 * Pull notification events from queue until stop time is reached
 	 *
-	 * @param int $stopTime The Unix time to stop sending notifications
-	 * @return int The number of notification events handled
+	 * @param int  $stopTime The Unix time to stop sending notifications
+	 * @param bool $matrix   If true, will return delivery matrix instead of a notifications event count
```

Comment:
What are the use cases for the matrix? I don't see it being used anywhere within the PR.

Without really knowing more, I'm wondering whether we should add `getDeliveryMatrix()` instead of having mixed return value. But that probably depends on the use cases?